drivers/clock_control/nrf5_power_clock.c

```diff
@@ -253,7 +269,16 @@ static void _power_clock_isr(void *arg)
 	done = (NRF_CLOCK->EVENTS_DONE != 0);
 	ctto = (NRF_CLOCK->EVENTS_CTTO != 0);
 
+#if defined(CONFIG_USB) && defined(CONFIG_SOC_NRF52840)
+	usb_detected = (NRF_POWER->EVENTS_USBDETECTED != 0);
```

Comment:
nrf_power.h (and other headers from ext/nordic/nrfx/hal) provides inline functions for handling register level operations in more readable way, i.e. nrf_power_event_check/clear.
It's OK for now; the target is to use nrfx drivers in most of shims.

Reply:
We can use nrfx if it's okay to mix with direct register read/write.
@cvinayak What do you think?

Reply:
Yes, please use nrfx interfaces, at least the inline ones should not be of any overhead.

Reply:
Okay.